### PR TITLE
setting the menu item order according to the webUI

### LINKED ISSUE #1749: Single file: overflow menu: same order as web ui
Web UI:
image

Android:
image

As far as I remember we said that those entries that are android only should be at bottom, @jancborchardt?

### PR DESCRIPTION
**Pull request by AndyScherzinger**
Fixes #1749

simply a reordering of the menu items.

cc @tobiasKaminsky @mario @ardevd for review

### CODE REVIEW

**Comment by mario** [comment]
👍 
orderInCategory 1 on everything *shrug*

**Comment by AndyScherzinger** [comment]
> orderInCategory 1 on everything shrug

@mario Just removed it via https://github.com/nextcloud/android/pull/2613/commits/cd3953b32fa963e63b7cb444505fcb34bda125b5 :+1:

**Comment by mario** [comment]
Thanks.

File: src/main/java/third_parties/daveKoeller/AlphanumComparator.java [hunk 1]
```diff
@@ -105,6 +105,11 @@ public int compare(T t1, T t2) {
     }
 
     public int compare(String s1, String s2) {
+        if ((s1 == null) || (s2 == null))
+        {
+            return 0;
```
Comment (tobiasKaminsky) [inline src/main/java/third_parties/daveKoeller/AlphanumComparator.java]: Why is this in this PR?
I am also not sure if we want to return "equal" (which is 0) when one of the two strings is null.
Please remove this and if needed, do it in a different PR. For sorting we have also test cases which then should be extended with "null" values. Currently we still have from time to time the TimSort problem, so any change should be decently tested.

**Comment by tobiasKaminsky** [comment]
:-1: 
Please remove the sorting stuff

File: src/main/java/third_parties/daveKoeller/AlphanumComparator.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/third_parties/daveKoeller/AlphanumComparator.java]: Stupid but done... have fun with the possible NPEs... not opening a PR for search + TimSort has oh so nothing to do with this since it'll crash anyways if one of the values is null... :roll_eyes:

Comment (AndyScherzinger) [reply src/main/java/third_parties/daveKoeller/AlphanumComparator.java]: > Why is this in this PR?
Because the actual implementation is obviously wrong? Dough....

**Comment by tobiasKaminsky** [comment]
:+1:
